agent.py

```diff
@@ -52,6 +55,28 @@ def pruneBuffer(self):
 		for key in buffer:
 			if (key > old_threshold):
 				buffer.pop(key)
+	
+	def tryPopBuffer(self):
+		#pops buffer and returns the story that is popped
+		#else, returns gracefully
+
+		if (len(buffer) > 0):
+			firstKey = sys.float_info.max
+			for key in buffer:
+				if key < firstKey:
+					firstKey = key
+			buffer.pop(firstKey)
```

Comment:
pop() removes and returns an item, so this section of the code would not work. (Related: realizing that coding without running to see it works is hard... we're gonna set up unit testing.)